### PR TITLE
dbt 1.7.1

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
New package for ❄️.

Note that we deviate a little bit in terms of dependencies. Most notably, the package using `protobuf` 3 instead of 4. From our analysis, we think this is safe to do and also saves us from a massive rebuild triggered by releasing and using protobuf 4 on defaults.

Upstream: https://github.com/dbt-labs/dbt-core/tree/v1.7.1

Depends on:
* https://github.com/AnacondaRecipes/dbt-semantic-interfaces-feedstock/pull/1
* https://github.com/AnacondaRecipes/dbt-extractor-feedstock/pull/5

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,56 +1,62 @@
...
 source:
   - url: https://pypi.io/packages/source/d/dbt-core/dbt-core-{{ version }}.tar.gz
-    sha256: 09c76b6b9bde3f8e00d24451a18af5b1c8151abfda9884603ce9d5bafc24c556
+    sha256: 78ac4711f35f7b7b1ad4bb17181b900ff9e15aed424e432e336fe110fec5981e
     folder: dbt-core
   - url: https://pypi.io/packages/source/d/dbt-postgres/dbt-postgres-{{ version }}.tar.gz
-    sha256: e6109322084316116d44717e57fb249edad8cf8ee05e83840a986c825cae07a9
+    sha256: cef8ffcf79d352d8cbc89a939753a546f2aad183688bd78743f385882bb9e001
     folder: dbt-postgres
+  - url: https://github.com/dbt-labs/dbt-core/archive/refs/tags/v{{ version }}.zip
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:14]: Trying something new here. I'm keeping the sources from PyPI and only downloading the tests from GitHub.

File: abs.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+channels:
+  - sfe1ed40
+  - https://staging.continuum.io/prefect/fs/dbt-semantic-interfaces-feedstock/pr1/cd87abc
```
Comment (JeanChristopheMorinPerso) [inline abs.yaml:3]: I will remove thse staging URLs as soon as the packages are available on sfe1ed40.

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Comment by skupr-anaconda** [comment]
@JeanChristopheMorinPerso Awesome! Thanks!

File: recipe/meta.yaml [hunk 3]
```diff
@@ -63,44 +70,64 @@ outputs:
         - dbt
         - dbt.adapters
       commands:
-        - pip check
+        # - pip check
```
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/meta.yaml:73]: because of protobuf

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,56 +1,63 @@
...
         - logbook >=1.5,<1.6
-        - click <9
+        - click >=8.0.2,<9
         - networkx >=2.3,<4
         - colorama >=0.3.9,<0.5
         - pathspec >=0.9,<0.12
         - isodate >=0.6,<0.7
         - sqlparse >=0.2.3,<0.5
-        - dbt-extractor >=0.4.1,<0.5.dev0
+        - dbt-extractor >=0.4.0,<0.6.dev0
```
Comment (cbouss) [inline recipe/meta.yaml:53]: Any reason for not setting this:
```suggestion
        - dbt-extractor >=0.5.0,<0.6.dev0
```
https://github.com/dbt-labs/dbt-core/blob/v1.7.1/core/setup.py#L74
It's difficult to tell from looking at this changelog. https://github.com/dbt-labs/dbt-extractor/blob/main/CHANGELOG.md

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:53]: No reason. It's probably a mistake from me. Good catch!

File: recipe/meta.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:73]: Exactly

File: abs.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+channels:
+  - sfe1ed40
+
```
Comment (JeanChristopheMorinPerso) [inline abs.yaml:3]:
```suggestion
```

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Pull request merged**